trieste/utils/pareto.py

```diff
@@ -101,10 +110,161 @@ def _bounds_2d(front: TensorType) -> BoundedVolumes:
+        ) -> Tuple[TensorType, TensorType, TensorType]:
+            divide_conquer_cells = tf.unstack(divide_conquer_cells, axis=0)
+            cell = divide_conquer_cells[-1]
+            divide_conquer_cells = tf.cond(
+                tf.not_equal(tf.size(divide_conquer_cells[:-1]), 0),
+                lambda: tf.stack(divide_conquer_cells[:-1]),
+                lambda: tf.zeros([0, 2, number_of_objectives], dtype=tf.int32),
+            )
+
+            arr = tf.range(number_of_objectives)
+            lower_idx = tf.gather_nd(pseudo_front_idx, tf.stack((cell[0], arr), -1))
+            upper_idx = tf.gather_nd(pseudo_front_idx, tf.stack((cell[1], arr), -1))
+            lower = tf.gather_nd(pseudo_front, tf.stack((lower_idx, arr), -1))
+            upper = tf.gather_nd(pseudo_front, tf.stack((upper_idx, arr), -1))
+
+            test_accepted = self._is_test_required((upper - jitter) < front)
+            lower_result, upper_result = tf.cond(
+                test_accepted,
+                lambda: self._accepted_test_body(lower_result, upper_result, lower_idx, upper_idx),
+                lambda: (lower_result, upper_result),
+            )
+
+            test_rejected = self._is_test_required((lower + jitter) < front)
+            divide_conquer_cells = tf.cond(
+                tf.logical_and(test_rejected, tf.logical_not(test_accepted)),
+                lambda: self._rejected_test_body(
+                    cell, lower, upper, divide_conquer_cells, total_size, threshold
+                ),
+                lambda: divide_conquer_cells,
+            )
+
+            return divide_conquer_cells, lower_result, upper_result
+
+        _, lower_result, upper_result = tf.while_loop(
+            lambda divide_conquer_cells, lower_result, upper_result: len(divide_conquer_cells) > 0,
+            while_body,
+            loop_vars=[divide_conquer_cells, lower_result, upper_result],
+            shape_invariants=[
+                tf.TensorShape([None, 2, number_of_objectives]),
+                tf.TensorShape([None, number_of_objectives]),
+                tf.TensorShape([None, number_of_objectives]),
+            ],
+        )
+        return BoundedVolumes(lower_result, upper_result)
+
+    def _accepted_test_body(
+        self,
+        lower_result: TensorType,
+        upper_result: TensorType,
+        lower_idx: TensorType,
+        upper_idx: TensorType,
+    ) -> Tuple[TensorType, TensorType]:
+        lower_result = tf.concat([lower_result, lower_idx[None]], axis=0)
+        upper_result = tf.concat([upper_result, upper_idx[None]], axis=0)
+        return lower_result, upper_result
+
+    def _rejected_test_body(
+        self,
+        cell: TensorType,
+        lower: TensorType,
+        upper: TensorType,
+        divide_conquer_cells: TensorType,
+        total_size: TensorType,
+        threshold: TensorType,
+    ) -> TensorType:
+
+        divide_conquer_cells_dist = cell[1] - cell[0]
```

Comment:
here's another re-assignment - I'm not quite sure what's happening here. You've got `divide_conquer_cells` as an argument then you immediately re-assign to it

Reply:
is this a bug?

Reply:
not quite sure on what you mean here if there is reassignment it should be at line 136? 
`divide_conquer_cells_dist` is different from `divide_conquer_cells`
could you elaborate?

Reply:
oh I misread this as `divide_conquer_cells` - all's fine